api/src/Command/NotificationEmails.php

```diff
@@ -49,42 +49,35 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             }
 
             $data = $user->getData();
-            $notif = isset($data['notification_emails']) ? $data['notification_emails'] : '';
+            $notif = isset($data['notification_emails']) ? $data['notification_emails'] : 'immediately';
+            $lastNotificationEmailCheck = $user->getLastNotificationEmailCheck();
+            $now = time();
+            if (empty($lastNotificationEmailCheck)) {
+                $user->setLastNotificationEmailCheck($now);
+                continue;
+            }
 
             // don't evaluate further if we are not in the right conditions
-            if (
-                'hourly' != $notif
-                && ('monthly' != $notif || '1-1' != date('j-G')) // first of the month at 1AM
-                && ('weekly' != $notif || '1-1' != date('N-G')) // monday at 1AM
-                && ('daily' != $notif || '1' != date('G')) // 1AM
-            ) {
+            $firstOfMonth1AM = strtotime('first day of this month 1:00 AM');
+            $lastMonday1AM = strtotime('last monday at 1:00 AM');
+            $today1AM = strtotime('today 1:00 AM');
+
+            $isHourlyDue = ('hourly' === $notif && (date('i', $now) === '00' || ($now - $lastNotificationEmailCheck) > 3600));
```

Comment:
Can we also use `strtotime` here ?

Reply:
Good call. On the hour instead of just each hour after the cron starts. I've added a commit to address this.